dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -1,571 +1,550 @@
+        weights=rules.tck2connectome.outputs.node_weights,
+        subcortical_seg=rules.add_brainstem_new_seg.output.seg,
+    output:
+        filter_mask=temp(
+            bids(
+                root=mrtrix_dir,
+                datatype="anat",
+                desc="exclude",
+                suffix="mask.mif",
+                **config["subj_wildcards"]
             )
-            combined_weights=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='subcortical',
-                suffix='tckweights.txt'
+        ),
+        filtered_tck=temp(
+            bids(
+                root=mrtrix_dir,
+                datatype="tractography",
+                desc="from{node1}to{node2}",
+                suffix="tractography.tck",
+                **config["subj_wildcards"]
             )
-        group: "subject_2"
-        container:
-            config['singularity']['mrtpipelines'],
-        shell:
-            'tckedit {input.tck} {output.combined_tck} '
-            'cat {input.weights} >> {output.combined_weights} '
-
-    rule filtered_connectome_map:
-        input:
-            weights=rules.combine_filtered.output.combined_weights,
-            tck=rules.combine_filtered.output.combined_tck, 
-            subcortical_seg=rules.add_brainstem_new_seg.output.seg,
-        params:
-            threads=workflow.threads,
-            radius=config['radial_search']
-        output:
-            sl_assignment=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='subcortical',
-                suffix='nodeassignment.txt',
-                **config['subj_wildcards'],
-            ),
-            node_weights=bids(
-                root=join(config['out_dir'], 'mrtpipelines'),
-                datatype='tractography',
-                space='T1w',
-                desc='subcortical',
-                suffix='nodeweights.csv'
-                **config['subj_wildcards'],
+        ),
+        filtered_weights=temp(
+            bids(
+                root=mrtrix_dir,
+                datatype="tractography",
+                desc="from{node1}to{node2}",
+                suffix="weights.csv",
+                **config["subj_wildcards"]
             )
-        group: "subject_2"
-        container:
-            config['singularity']['mrtpipelines']
-        shell:
-            'tck2connectome -nthreads {params.threads} -zero_diagonal -stat_edge sum -assignment_radial_search {params.radius} -tck_weights_in {input.weights} -out_assignments {output.sl_assignment} -symmetric {input.tck} {input.subcortical_seg} {output.node_weights} -force'
\ No newline at end of file
+        ),
+    threads: workflow.cores
+    group:
+        "subject_2"
+    container:
+        config["singularity"]["mrtrix"]
+    shell:
```

Comment:
I don't know this part of mrtrix very well but seems like this should have a `&&` between the first and second lines (or split into two rules)?